client/service_update.go

```diff
@@ -33,13 +33,29 @@ func (cli *Client) ServiceUpdate(ctx context.Context, serviceID string, version
 
 	query.Set("version", strconv.FormatUint(version.Index, 10))
 
+	// Call the /manifest endpoint to retrieve manifest information.
+	// This happen only when the image has changed.
+	var manifestErr error
+	if options.UpdateImage {
+		manifestInspect, manifestErr := cli.ImageManifest(ctx, service.TaskTemplate.ContainerSpec.Image, options.EncodedRegistryAuth)
+		if manifestErr == nil {
+			// manifest was successfully retrieved, now pin by digest
+			service.TaskTemplate.ContainerSpec.Image = string(manifestInspect.Digest)
```

Comment:
We must not overwrite the `Image` field here if it already contains a digest (for example, if digest pinning was already done through content trust).